### PR TITLE
20488: Declare <Chip/> as deprecated

### LINKED ISSUE #20488: Deprecate Chip in favor of Tag
### Description

To prevent further tech debt and inconsistencies `Chip` can be deprecated in favor of `Tag`

`Chip` (old): `ui/components/ui/chip/chip.js`
`Tag` (new): `ui/components/component-library/tag/tag.tsx`

Add the deprecation JSDoc format below to the deprecated code

```
/**
 * @deprecated The `<Chip />` component has been deprecated in favor of the new `<Tag>` component from the component-library.
 * Please update your code to use the new `<Tag>` component instead, which can be found at ui/components/component-library/tag/tag.tsx.
 * You can find documentation for the new `Tag` component in the MetaMask Storybook:
 * {@link https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-tag--docs}
 * If you would like to help with the replacement of the old `Chip` component, please submit a pull request against this GitHub issue:
 * {@link https://github.com/MetaMask/metamask-extension/issues/20487}
 */
```

### Technical Details

- Add deprecation notice to storybook
- Add deprecation message above old component function or class
- Create good first issue to replace all old components with new [Replace old Chip component with new Tag component](https://github.com/georgewrmarshall/issue-templates/blob/main/REPLACE.md)

### Acceptance Criteria

- Old component has been deprecated (deprecation indicator shows up in VS Code)
- Good first issue has been created for the community to contribute to reducing tech debt and improving the UI consistency of MetaMask

If the acceptance criteria is not met, PRs may be closed.

### Difficulty: Intermediate

Good first issue for: External contributors who are able to run the extension locally, familiar with React, component props and proficient at searching a codebase

### PR DESCRIPTION
**Pull request by jase-b**
## Explanation

Fixes: #20488

`<Chip/>` has been deprecated in favor of `<Tag/>`.

Warnings and JSDocs for deprecation have been applied to instances of `<Chip/>`.

### Before

VSCode Detection:
20488-jsdoc-before

Storybook:
20488-story-before

### After

VSCode Detection:
20488-jsdoc-after

Storybook:
20488-story-after

## Manual Testing Steps

**VSCode**
1. Search for '<Chip' in the MM codebase.
2. Verify at least one implementation of `<Chip/>` shows a strikethrough to indicate deprecation.

**Storybook**
1. Run `yarn storybook`
2. In the sidebar under 'Components', click `UI` -> `Chip`.
3. Verify all iterations of `Chip` have a 'Deprecated' banner message.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! This is fantastic @jase-b thanks for your contribution!

File: ui/components/ui/chip/chip.js [hunk 1]
```diff
@@ -11,6 +11,14 @@ import {
...
 } from '../../../helpers/constants/design-system';
 
+/**
+ * @deprecated The `<Chip />` component has been deprecated in favor of the new `<Tag>` component from the component-library.
+ * Please update your code to use the new `<Tag>` component instead, which can be found at ui/components/component-library/tag/tag.tsx.
+ * You can find documentation for the new `Tag` component in the MetaMask Storybook:
+ * {@link https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-tag--docs}
+ * If you would like to help with the replacement of the old `Chip` component, please submit a pull request against this GitHub issue:
+ * {@link https://github.com/MetaMask/metamask-extension/issues/20487}
+ */
```
Comment (georgewrmarshall) [inline ui/components/ui/chip/chip.js:21]: Adding a line break here should fix the linting error
```suggestion
 */
 
```

**Review by georgewrmarshall — COMMENTED** [review]
One suggestion to fix the linting error

**Review by georgewrmarshall — APPROVED** [review]
LGTM!

**Review by darkwing — APPROVED** [review]

**Comment by darkwing** [comment]
Thank you @jase-b !

**Pull request merged**